### ISSUE TITLE
Mixture of heightmap plus procedural details?

### ISSUE THREAD

**Issue by bjfar** [mohsenph69/Godot-MTerrain-plugin#58]
This is not an issue, just a question. This plugin looks amazing and I commend your efforts, I am just curious whether you think it is a suitable basis for an experiment I want to try. I am interested in combining low-resolution terrain data with high-resolution procedurally generated details in order to create extremely large worlds without needing terabytes of storage. For example I have a global Mars heightmap dataset at approx 200m resolution, which is 10GB, but of course that looks bad at 1m "player scale" resolution. So I'd like to dynamically add some high-frequency noise or whatever to make things look better on small scales. Doesn't matter what it is exactly, just something generated at runtime only nearby to the player, of low enough amplitude to be irrelevant at large distances.

I know this is not what you have in mind with this plugin, but do you think it is at all compatible?

**Comment by mohsenph69** [comment]
So to adding detail with your detail map to Heightmap! you need to have enough vertices on Terrain! Right now this things in MTerrain is combined! I mean when create a heightmap for terrain! terrain plugin will match exactly each pixel of heightmap to one vertex!
So to do what you want we need to tweak some stuff in terrain! which that is not to difficult to do do! but it take time! I will consider that!

**Comment by bjfar** [comment]
Yeah that sounds good, maybe add some sort of grid resolution option to generate (some of) the meshes at resolution higher than the heightmaps, and just interpolate the heightmap onto it. Then users can modify the vertices further if desired. 

No worries re. time, let's call it a feature request then :).
